message_app_client/src/components/conversation_panel/ConversationCard.js

```diff
@@ -9,6 +10,19 @@ function ConversationCard(props){
     // props.conversationID: int
 
     const [otherParticipantStatus, setOtherParticipantStatus] = useState(props.otherParticipantStatus);
+    const { conversationID, setConversationID } = useContext(ConversationDataContext)
+
+    const joinConversation = () => {
+        if (props.conversationID !== conversationID){
+            let oldConversationID = conversationID;
+            setConversationID(props.conversationID)
+            const payload = {
+                oldConversationID: oldConversationID,
+                newConversationID: props.conversationID
+            }
```

Comment:
Each `conversationCard` will have one and only one constant conversationID, right? Why does it even change here? Did I miss something?

Reply:
The if statement is used to block the user from joining one conversation multiple times.

Reply:
Using the if statement as a guarding condition would be more readable. Something like this
```JavaScript
// if a user try to join a conversation they are already in, do nothing
if (props.conversationID === conversationID) return;

let oldConversationID = conversationID;
setConversationID(props.conversationID);
const payload = {
    oldConversationID: oldConversationID,
    newConversationID: props.conversationID
};
socket.emit("join_conversation", payload);

```